moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -709,6 +712,98 @@ TEST_F(OneRobot, testRelativeJointSpaceJump)
   EXPECT_NEAR(1.0, fraction, 0.01);
 }
 
+class PR2 : public testing::Test
+{
+protected:
+  std::string readFileToString(boost::filesystem::path path) const
+  {
+    std::string file_string;
+    std::fstream file(path.string().c_str(), std::fstream::in);
+    if (file.is_open())
+    {
+      std::string line;
+      while (file.good())
+      {
+        std::getline(file, line);
+        file_string += (line + "\n");
+      }
+      file.close();
+    }
+    return file_string;
+  }
+
+  void SetUp() override
+  {
+    boost::filesystem::path res_path(MOVEIT_TEST_RESOURCES_DIR);
+    const std::string urdf = this->readFileToString(res_path / "pr2_description/urdf/robot.xml");
+    const std::string srdf = this->readFileToString(res_path / "pr2_description/srdf/robot.xml");
+
+    urdf::ModelInterfaceSharedPtr parsed_urdf(urdf::parseURDF(urdf));
+    if (!parsed_urdf)
+      throw std::runtime_error("Cannot parse URDF.");
+
+    srdf::ModelSharedPtr parsed_srdf(new srdf::Model());
+    bool srdf_ok = parsed_srdf->initString(*parsed_urdf, srdf);
+    if (!srdf_ok)
+      throw std::runtime_error("Cannot parse URDF.");
+
+    model_.reset(new robot_model::RobotModel(parsed_urdf, parsed_srdf));
+  };
+
+  void TearDown() override
+  {
+  }
+
+  robot_model::RobotModelPtr model_;
+};
+
+TEST_F(PR2, StateUpdateTiming)
+{
+  ASSERT_TRUE(bool(model_));
+  robot_state::RobotState state(model_);
+  clock_t begin = clock();
+  for (unsigned i = 0; i < 10000; ++i)
+  {
+    state.setToRandomPositions();
+    state.update();
+  }
+  clock_t end = clock();
+  double elapsed_secs = double(end - begin) / CLOCKS_PER_SEC;
+  std::cerr << "time for 10,000 random RobotState updates: " << elapsed_secs << "s" << std::endl;
```

Comment:
this isn't really a test, but rather a benchmark. having this run 10000 times will slow down the build without any useful outcome. can this instead just be a helper function or maybe we comment it out?